utils.js

```diff
@@ -6636,38 +6674,15 @@ export const extractJarArchive = function (
       if (jarResult.status !== 0) {
         console.error(jarResult.stdout, jarResult.stderr);
       } else {
-        let group = "",
-          name = "",
-          version = "",
-          confidence = 1,
-          technique = "manifest-analysis";
         // When maven descriptor is available take group, name and version from pom.properties
         // META-INF/maven/${groupId}/${artifactId}/pom.properties
         // see https://maven.apache.org/shared/maven-archiver/index.html
-        if (existsSync(mavenDir)) {
-          let groupDir = readdirSync(mavenDir);
-          if (groupDir && groupDir.length) {
-            let artifactDir = readdirSync(join(mavenDir, groupDir[0]));
-            if (artifactDir && artifactDir.length) {
-              let pomPropertiesFile = join(
-                mavenDir,
-                groupDir[0],
-                artifactDir[0],
-                "pom.properties"
-              );
-              if (existsSync(pomPropertiesFile)) {
-                const pomProperties = parsePomProperties(
-                  readFileSync(pomPropertiesFile, {
-                    encoding: "utf-8"
-                  })
-                );
-                group = pomProperties["groupId"];
-                name = pomProperties["artifactId"];
-                version = pomProperties["version"];
-              }
-            }
-          }
-        }
+        const pomProperties = getPomPropertiesFromMavenDir(mavenDir);
```

Comment:
Since pomProperties can be an empty object, do we need a condition to skip?

Reply:
Apologies, it might be right below this.

Reply:
If it is an empty object (e.g. wsdl4j-1.6.3.jar) group, name and version will be undefined, but this case is handled later.